src/ApiException.php

```diff
@@ -48,13 +49,31 @@ public function __construct($message, $code, \Exception $previous = null)
      */
     public static function createFromStdClass($status)
     {
-        $ex = new ApiException($status->details, $status->code);
-        if (property_exists($status, 'metadata')) {
-            $ex->metadata = $status->metadata;
-        }
+        $basicMessage = $status->details;
+        $code = $status->code;
+        $metadata = property_exists($status, 'metadata') ? $status->metadata : null;
+
+        $messageData = [
+            'message' => $basicMessage,
+            'code' => $code,
+            'status' => GrpcConstants::getStatusNameFromCode($status->code),
+            'details' => Serializer::decodeMetadata($metadata)
+        ];
+
+        $message = json_encode($messageData, JSON_PRETTY_PRINT);
+
+        $ex = new ApiException($message, $code);
+        $ex->basicMessage = $basicMessage;
+        $ex->metadata = $metadata;
```

Comment:
No, and I think this is not a great pattern. I have added the `metadata` and `basicMessage` fields as optional params in the constructor.